### ISSUE TITLE
Need support of FastLayerNorm

### ISSUE THREAD

**Issue by yidong72** [NVIDIA/apex#1292]
Currently the method `_get_params_for_weight_decay_optimization` doesn't support `FastLayerNorm` while NeMo Megaton uses it. I fix it by the following code. We need it in the apex library. 
```python
def _get_params_for_weight_decay_optimization(
        model: Union[torch.nn.Module, List[torch.nn.Module]],
) -> Dict[str, torch.nn.Parameter]:
    """Divide params into with-weight-decay and without-weight-decay groups.

    Layernorms and biases will have no weight decay but the rest will.
    """
    modules = listify_model(model)
    weight_decay_params = {'params': []}
    no_weight_decay_params = {'params': [], 'weight_decay': 0.0}
    for module in modules:
        for module_ in module.modules():
            if isinstance(module_, (FusedLayerNorm, FastLayerNorm)):
                no_weight_decay_params['params'].extend(
                    [p for p in list(module_._parameters.values())
                     if p is not None])
            else:
                weight_decay_params['params'].extend(
                    [p for n, p in list(module_._parameters.items())
                     if p is not None and n != 'bias'])
                no_weight_decay_params['params'].extend(
                    [p for n, p in list(module_._parameters.items())
                     if p is not None and n == 'bias'])

    return weight_decay_params, no_weight_decay_params
```

**Comment by crcrpar** [comment]
I'm wondering if NeMo can have a similar function in it as this function doesn't require any apex internal information. I didn't expect this function is used outside  (which is why I prefixed this with an underscore)

**Comment by eqy** [comment]
I believe the proposal from @crcrpar is to remove this function from apex or better indicate that it is internal (e.g., move it to testing module/namespace), since its implementation has no internal dependencies in apex and it is very possible that future users of new normalization layers would want to modify this function independently.

**Comment by crcrpar** [comment]
yeah, to keep this function au courant with the usage outside apex is hard for both of us I think.
Also if you have one in your side, keep it there.

The change in my mind is the linked PR.
